fondant/component.py

```diff
@@ -223,17 +224,70 @@ def transform(self, *args, **kwargs) -> dd.DataFrame:
             kwargs: Arguments provided to the component are passed as keyword arguments
         """
 
+    @abstractmethod
     def _process_dataset(self, manifest: Manifest) -> dd.DataFrame:
         """
-        Creates a DataLoader using the provided manifest and loads the input dataframe using the
-        `load_dataframe` instance, and  applies data transformations to it using the `transform`
-        method implemented by the derived class. Returns a single dataframe.
+        Load the data based on the manifest using a DaskDataloader and call the transform method to
+        process it.
 
         Returns:
             A `dd.DataFrame` instance with updated data based on the applied data transformations.
         """
         data_loader = DaskDataLoader(manifest=manifest, component_spec=self.spec)
         df = data_loader.load_dataframe()
         df = self.transform(dataframe=df, **self.user_arguments)
+        return df
+
+
+class PandasTransformComponent(TransformComponent):
+    @abstractmethod
+    def setup(self, *args, **kwargs):
+        """Called once for each instance of the Component class. Use this to set up resources
+        such as database connections.
+        """
+
+    def _process_dataset(self, manifest: Manifest) -> dd.DataFrame:
+        """
+        Load the data based on the manifest using a DaskDataloader and call the transform method to
+        process it.
+
+        Returns:
+            A `dd.DataFrame` instance with updated data based on the applied data transformations.
+        """
+        data_loader = DaskDataLoader(manifest=manifest, component_spec=self.spec)
+        df = data_loader.load_dataframe()
+
+        self.setup(**self.user_arguments)
```

Comment:
This takes in all the arguments passed to the components which some of them might be related to the setup while other can be related to the custom transform.

Reply:
These arguments are at the component level, so it makes sense to pass them in during setup. If the user wants to leverage them in the `transform` function, they can register them as instance attributes in the setup method (`self.arg = arg`) and use them in the `transform` method.

Reply:
alright so if I understand correctly this is mainly to avoid re-initializing arguments everytime you execute `transform` on a partition? A bit similar to how Beam does it where you have DoFn.setup() and DoFn.process() (equivalent to our transform)